components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -195,6 +195,7 @@ pub fn get_cast_fn_rpn_node(
         args_len: 1,
         field_type: to_field_type,
         implicit_args: Vec::new(),
+        metadata: Box::new(()),
```

Comment:
Will `Box<()>` allocate?

Reply:
No. Box::new(()) always points to `0x1` on Linux x86_64